src/main/java/vct/col/veymont/StructureCheck.scala

```diff
@@ -0,0 +1,451 @@
+package vct.col.veymont
+
+import hre.lang.System.Fail
+import vct.col.ast.`type`.{ClassType, PrimitiveSort, PrimitiveType, Type}
+import vct.col.ast.expr.{MethodInvokation, NameExpression, OperatorExpression, StandardOperator}
+import vct.col.ast.generic.ASTNode
+import vct.col.ast.stmt.composite.{BlockStatement, IfStatement, LoopStatement, ParallelRegion}
+import vct.col.ast.stmt.decl.Method.{JavaConstructor, Kind}
+import vct.col.ast.stmt.decl._
+import vct.col.ast.stmt.terminal.AssignmentStatement
+import vct.col.ast.util.ASTUtils
+import Util._
+import vct.col.veymont.StructureCheck.{getRoleOrHelperClass, isResourceType}
+
+import scala.collection.convert.ImplicitConversions.{`collection asJava`, `iterable AsScalaIterable`}
+
+object StructureCheck {
+  def getMainClass(source : ProgramUnit) : ASTClass = source.get().find(_.name == mainClassName).get.asInstanceOf[ASTClass]
+  def getRoleOrHelperClass(source : ProgramUnit) =
+    source.get().filter(c => c.isInstanceOf[ASTClass] && isRoleOrHelperClassName(c.name)).map(_.asInstanceOf[ASTClass])
+  def getRoleNames(source : ProgramUnit) : Iterable[String] =
+    getMainClass(source).methods().find(_.kind== Kind.Constructor) //getMainConstructor
+      .get.getBody.asInstanceOf[BlockStatement].getStatements.map(_.asInstanceOf[AssignmentStatement]) //getRoleObjects
+      .map(_.location.asInstanceOf[NameExpression].name) //getRoleNames
+
+  def isExecutableMainMethod(m : Method) : Boolean = m.name != mainMethodName && m.kind != Method.Kind.Constructor && m.kind != Method.Kind.Pure && !(m.kind == Method.Kind.Predicate && isResourceType(m.getReturnType))
+
+  private def isResourceType(t : Type) : Boolean = t match {
+    case p : PrimitiveType => p.sort == PrimitiveSort.Resource
+    case _ => false
+  }
+
+  def isAllowedPrimitive(p : PrimitiveType) = p.isBoolean || p.isDouble || p.isInteger
+}
+
+class StructureCheck(source : ProgramUnit) {
+
+  private var mainClass : ASTClass = null
+  private var roleObjects : Iterable[AssignmentStatement] = null
+  private var roleNames : Iterable[String] = null
+  private var roleClasses : Iterable[ASTClass] = null
+  private var roleClassNames : Iterable[String] = null
+  private var mainMethods : Iterable[Method] = null
+  private var mainMethodNames : Iterable[String] = null
+  private var nonPlainMainMethodNames : Iterable[String] = null
+  private var otherClasses : Iterable[ASTClass] = null
+  private var prevAssertArg : ASTNode = null
+
+  def check() : Unit = {
```

Comment:
I think this `check` method is the main entry point/interface of `StructureCheck`, yes? I would then just initialize them in the constructor (by which I mean the area where you declare all your `var`s; in scala you can write arbitrary code there.) That way you can avoid the situation where you forget to initialize a nulled value, and also do not need a million `.get`s if you make them options.